File: cashier/src/main/java/com/secwager/cashier/di/CashierModule.java
package com.secwager.cashier.di;

import com.zaxxer.hikari.HikariConfig;
import com.zaxxer.hikari.HikariDataSource;
import dagger.Module;
import dagger.Provides;

import javax.inject.Singleton;
import javax.sql.DataSource;
import org.apache.commons.dbutils.QueryRunner;
import org.postgresql.ds.PGPoolingDataSource;

import java.util.Properties;

@Module
public class CashierModule {



  @Provides
  @Singleton
  public DataSource provideDataSource() {
    HikariConfig config = new HikariConfig();
    config.setAutoCommit(false);
    config.setDataSourceClassName(org.postgresql.ds.PGSimpleDataSource.class.getName());
    config.setJdbcUrl(System.getenv("JDBC_URL"));
    config.setUsername(System.getenv("DB_USER"));
    config.setUsername(System.getenv("DB_PASSWORD"));
    HikariDataSource ds = new HikariDataSource(config);
    return ds;
  }


  @Provides
  @Singleton
  public QueryRunner provideQueryRunner(DataSource dataSource){
    return new QueryRunner(dataSource);
  }




}

File: cashier/src/main/kotlin/di/CashierModule.kt
package com.secwager.cashier.di

import com.github.jasync.sql.db.Connection
import com.github.jasync.sql.db.postgresql.PostgreSQLConnectionBuilder
import com.secwager.dao.cashier.CashierDao
import com.secwager.dao.cashier.CashierDaoJasyncImpl

import dagger.Module
import dagger.Provides
import java.util.*
import javax.inject.Singleton
import javax.sql.DataSource


@Module
class CashierModule {
    @Provides
    @Singleton
    fun provideConnectionPool(): Connection {
        return PostgreSQLConnectionBuilder.createConnectionPool {
            password = ""
            host = ""
            username = ""
            database = "secwager"
            connectionCreateTimeout = 10000
            port = 77
        }
    }


    @Provides
    @Singleton
    fun provideCashierDao(connection: Connection): CashierDao {
        return CashierDaoJasyncImpl(connection)
    }


}